tsdb/shard.go

```diff
@@ -702,56 +702,64 @@ func (s *Shard) validateSeriesAndFields(points []models.Point, tracker StatsTrac
 			continue
 		}
 
-		name := p.Name()
-		mf := engine.MeasurementFields(name)
-
-		// Check with the field validator.
-		if err := ValidateFields(mf, p, s.options.Config.SkipFieldSizeValidation); err != nil {
-			switch err := err.(type) {
-			case PartialWriteError:
-				if reason == "" {
-					reason = err.Reason
+		cont, err := func(p models.Point, iter models.FieldIterator) (cont bool, err error) {
```

Comment:
It looks like the only time we don't have `cont == true` is where there is an error, and then the `continue` is at the bottom of the loop. Is `cont` really doing anything?